File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/type/TypeService.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.type;

import java.util.List;
import java.util.Map;

public interface TypeService {

    List<Type> findAll();

    Type get(Integer id);

    Integer create(Type type);

    void update(Integer id, Type type);

    void delete(Integer id);

    Map<Integer, Integer> getTypeValues();
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/type/TypeService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.type

interface TypeService {

    fun findAll(): List<Type>

    fun `get`(id: Int): Type

    fun create(type: Type): Int

    fun update(id: Int, type: Type)

    fun delete(id: Int)

    fun getTypeValues(): Map<Int, Int>
}
